docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+----
+--
+====
+
+== Manage data
+All methods of `VectorCollection` that work with collection data are asynchronous. The result is returned as a `CompletionStage`. A collection interacts with objects in the form of documents (`VectorDocument`). Each document comprises a value and one or more vectors associated with that value.
+
+=== Create document
+To create a document use statics the methods of `VectorDocument` and `VectorValues` classes.
+
+Example single index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of(
+        "document value",
+        VectorValues.of(
+                new float[]{0.2f, 0.9f, -1.2f}
+        )
+);
+----
+--
+====
+
+For multi-index collections, specify the names of the indexes to which the vectors belong:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of(
+        "some document value",
+        VectorValues.of(
+                Map.of(
+                        "index1", new float[] {1.2f, -0.3f, 2.2f},
+                        "index2", new float[] {0.2f, 1.3f, -0.2f}
+                )
+        )
+);
+----
+--
+====
+
+
+=== Put objects
+To put single document to a vector collection, use the method `putAsync`, `putIfAbsent` or `setAsync` of the `VectorCollection` class.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+VectorDocument<String> document = VectorDocument.of("document value", VectorValues.of(new float[] {0.2f, 0.9f, -1.2f}));
+CompletionStage<VectorDocument<String>> result = vectorCollection.putAsync("1", document);
+----
+--
+====
+
+To put several documents to a vector collection, use the method `putAllAsync` of the `VectorCollection` class.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+VectorDocument<String> document1 = VectorDocument.of("some document1 value", VectorValues.of(new float[] {0.2f, 0.9f, -1.2f}));
+VectorDocument<String> document2 = VectorDocument.of("some document2 value", VectorValues.of(new float[] {1.2f, -0.3f, 2.2f}));
+CompletionStage<Void> result = vectorCollection.putAllAsync(Map.of("1", document1, "2", document2));
+----
+--
+====
+
+=== Read objects
+To get document from a vector collection, use the method `getAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<VectorDocument<String>> result = vectorCollection.getAsync("1");
+----
+--
+====
+
+=== Update objects
+To update single document in a vector collection, use the method `putAsync` or `setAsync` of the `VectorCollection` class.
```

Comment:
"Update objects", "update single document" - we should use consistent names

Reply:
b2fd6c714396ed311c966efd372253f1d485ff18